backend/core/utils/tool.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python
+import time
+
+import psycopg2.extras
+
+ISOLEVEL = psycopg2.extensions.ISOLATION_LEVEL_AUTOCOMMIT
+
+LIMIT_RETRIES = 5
+
+
+class DB:
+    def __init__(
+        self,
+        user,
+        password,
+        host,
+        port,
+        database,
+        reconnect=True,
+        autocommit=False,
+        readonly=False,
+    ):
+        self.user = user
+        self.password = password
+        self.host = host
+        self.port = port
+        self.database = database
+        self._connection = None
+        self._cursor = None
+        self.reconnect = reconnect
+        self.autocommit = autocommit
+        self.readonly = readonly
+        self.init()
+
+    def connect(self, retry_counter=0):
+        if not self._connection:
+            try:
+                self._connection = psycopg2.connect(
+                    user=self.user,
+                    password=self.password,
+                    host=self.host,
+                    port=self.port,
+                    database=self.database,
+                    connect_timeout=3,
+                )
+                retry_counter = 0
+                self._connection.set_session(
+                    autocommit=self.autocommit, readonly=self.readonly
+                )
+                return self._connection
+            except psycopg2.OperationalError as error:
```

Comment:
what about psycopg2.DatabaseError?

Reply:
don't think it's needed for this part. it's when trying to execute mostly
http://initd.org/psycopg/docs/errors.html